### PR TITLE
Add ResetVotes button.

### PR DESCRIPTION
**Pull request by jhb188**
screen shot 2017-12-28 at 10 54 58 pm

### CODE REVIEW

File: src/components/SongPage/SongList/Song/Upvote.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { isUpvoted } from 'util/votes'
 const Upvote = ({changeVote, songId}) => {
   const styles = {
     arrowStyles: {
+      cursor: 'pointer',
```
Comment (jhb188) [inline src/components/SongPage/SongList/Song/Upvote.js]: added some `cursor: 'pointer'`s to make these look clickable

File: src/components/SongPage/SongList/Song/Visibility/Visibility.js [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 
-const Visibility = ({visible = true, changeVisible, firebase}) => {
-  if (firebase.auth().currentUser) {
-    if (!visible) return <div style={styles.isHidden} onClick={() => {changeVisible(true)}}>Show</div>
-    return <div onClick={() => {changeVisible(false)}}>Hide</div>
-  }
-  return <div></div>
-
-}
+const Visibility = ({visible = true, changeVisible}) =>
```
Comment (jhb188) [inline src/components/SongPage/SongList/Song/Visibility/Visibility.js]: wrapped this in the `showIfAuthenticated` HOC and tried to condense the render

File: src/util/showIfAuthenticated.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      return <DecoratedComponent { ...otherProps } />
+    }
+
+    return <div />
+  }
+
+  return firebaseConnect([
+    'auth'
+  ])(OnlyShowIfAuthenticated)
+}
```
Comment (jhb188) [inline src/util/showIfAuthenticated.js]: saw we would be reusing this logic, so I pulled it into a HOC so we can just decorate whichever components need it. was thinking we may also want a version of this that shows an alternative component when not authenticated.

Comment (rossvz) [reply src/util/showIfAuthenticated.js]: I love it! HOCs are still a little new for me, but I knew there had to be a "react" way of handling this sort of logic! So clean

**Pull request merged**